logic.js

```diff
@@ -279,3 +279,37 @@ document.addEventListener('DOMContentLoaded', () => {
     validateForm();
   });
 });
+document.addEventListener('DOMContentLoaded', () => {
+  function preserveChanges(myField, myChange) {
+    const data = localStorage.getItem('userData');
+    const tmpData = JSON.parse(data);
+    tmpData[myField] = myChange;
+    const forSave = JSON.stringify(tmpData);
+    localStorage.setItem('userData', forSave);
+  }
+  const myNameInput = document.getElementById('fname');
+  myNameInput.addEventListener('input', () => {
+    preserveChanges('userName', myNameInput.value);
+    return 0;
+  });
+  const myEmailInput = document.getElementById('faddress');
+  myEmailInput.addEventListener('input', () => {
+    preserveChanges('email', myEmailInput.value);
+    return 0;
+  });
+});
+
+window.onload = function initDataPreserv() {
+  const userData = localStorage.getItem('userData');
+  if (userData === null) {
+    const createData = { userName: 'FullName', email: 'Email Address' };
+    const storeD = JSON.stringify(createData);
+    localStorage.setItem('userData', storeD);
+  } else {
+    const tmpData = JSON.parse(userData);
+    const myNameInput = document.getElementById('fname');
+    myNameInput.value = tmpData.userName;
+    const myEmailInput = document.getElementById('faddress');
+    myEmailInput.value = tmpData.email;
+  }
+};
```

Comment:
- [ ] Well done on saving the user's name and email in the local storage! However, I noticed that you're not currently saving the content from the message input field. Just a friendly reminder that this is a [project requirement](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/10_localstorage.md), and **it would greatly enhance the user's experience if they could store their message even after reloading the page.** Keep up the great work! 🙌

   ![image](https://github.com/alexansaa/Personal_Portafolio/assets/107952724/35e1c6ef-a4f6-404f-9b2a-eccc75535d63)